internal/serve/httphandler/wallet_creation_handler.go

```diff
@@ -0,0 +1,119 @@
+package httphandler
+
+import (
+	"encoding/hex"
+	"net/http"
+	"strings"
+
+	"github.com/stellar/go/support/http/httpdecode"
+	"github.com/stellar/go/support/render/httpjson"
+
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/data"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/serve/httperror"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/serve/validators"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/services"
+	"github.com/stellar/stellar-disbursement-platform-backend/stellar-multitenant/pkg/tenant"
+)
+
+type WalletCreationHandler struct {
+	embeddedWalletService services.EmbeddedWalletServiceInterface
+}
+
+type CreateWalletRequest struct {
+	Token     string `json:"token"`
+	PublicKey string `json:"public_key"`
+}
+
+func (r CreateWalletRequest) Validate() *httperror.HTTPError {
+	validator := validators.NewValidator()
+	validator.Check(len(r.Token) > 0, "token", "token should not be empty")
+	validator.Check(len(r.PublicKey) > 0, "public_key", "public_key should not be empty")
+	if _, err := hex.DecodeString(r.PublicKey); err != nil {
+		validator.AddError("public_key", "public_key should be a valid hex string")
+	}
+
+	if validator.HasErrors() {
+		return httperror.BadRequest("", nil, validator.Errors)
+	}
+
+	return nil
+}
+
+type WalletResponse struct {
+	ContractAddress string                    `json:"contract_address,omitempty"`
+	Status          data.EmbeddedWalletStatus `json:"status"`
+}
+
+func (h WalletCreationHandler) getTenantFromContext(ctx http.Request) (*tenant.Tenant, *httperror.HTTPError) {
+	currentTenant, err := tenant.GetTenantFromContext(ctx.Context())
+	if err != nil || currentTenant == nil {
+		return nil, httperror.InternalError(ctx.Context(), "Failed to load tenant from context", err, nil)
+	}
+	return currentTenant, nil
+}
```

Comment:
I don't think you need this method, you're not making the code shorter nor adding any extra functionality.

FYI: if the tenant is not in the context, the method already returns an error:
https://github.com/stellar/stellar-disbursement-platform-backend/blob/3b6bd647f115876b7ecdd5e240ae81253427f272/stellar-multitenant/pkg/tenant/manager.go#L356-L363